console/frontend/src/main/frontend/src/app/components/datatable/datatable.component.ts

```diff
@@ -0,0 +1,273 @@
+import { CdkTableModule, DataSource } from '@angular/cdk/table';
+import { CommonModule } from '@angular/common';
+import { BehaviorSubject, Observable, Subscription } from 'rxjs';
+import { FormsModule } from '@angular/forms';
+import { DtContentDirective } from './dt-content.directive';
+import { TruncatePipe } from '../../pipes/truncate.pipe';
+import { ToDateDirective } from '../to-date.directive';
+
+export type TableOptions = {
+  sizeOptions: number[];
+  size: number;
+  serverSide: boolean;
+  filter: boolean;
+};
+
+export type DataTableColumn<T> = {
+  name: string;
+  displayName: string;
+  property: keyof T | null;
+
+  html?: boolean;
+  className?: string;
+  hidden?: boolean;
+};
+
+export type DataTableEntryInfo = {
+  minPageEntry: number;
+  maxPageEntry: number;
+  totalFilteredEntries: number;
+  totalEntries: number;
+};
+
+export type DataTableServerRequestInfo = {
+  size: number;
+  offset: number;
+  sort: 'asc' | 'desc';
+};
+
+export type DataTableServerResponseInfo<T> = {
+  totalEntries: number;
+  filteredEntries: number;
+  size: number;
+  offset: number;
+  data: T[];
+};
+
+@Component({
+  selector: 'app-datatable',
+  standalone: true,
+  imports: [CommonModule, FormsModule, CdkTableModule, TruncatePipe, ToDateDirective, ToDateDirective],
+  templateUrl: './datatable.component.html',
+  styleUrl: './datatable.component.scss',
+})
+export class DatatableComponent<T> implements AfterViewInit, OnDestroy {
+  @Input({ required: true }) public datasource!: DataTableDataSource<T>;
+  @Input({ required: true }) public displayColumns: DataTableColumn<T>[] = [];
+  @Input() public truncate: boolean = false;
+  @Input() public truncateLength: number = 30;
+
+  @ContentChild(DtContentDirective) protected content!: DtContentDirective<T>;
+  protected totalFilteredEntries: number = 0;
+  protected totalEntries: number = 0;
+  protected minPageEntry: number = 0;
+  protected maxPageEntry: number = 0;
+
+  protected get displayedColumns(): string[] {
+    return this.displayColumns.map((column) => column.name);
+  }
+
+  private datasourceSubscription: Subscription = new Subscription();
+
+  ngAfterViewInit(): void {
+    if (this.datasource) {
+      const subscription = this.datasource.getEntriesInfo().subscribe((entriesInfo) => {
+        this.totalEntries = entriesInfo.totalEntries;
+        this.totalFilteredEntries = entriesInfo.totalFilteredEntries;
+        this.minPageEntry = entriesInfo.minPageEntry;
+        this.maxPageEntry = entriesInfo.maxPageEntry;
+      });
+      this.datasourceSubscription.add(subscription);
+    }
+  }
+
+  ngOnDestroy(): void {
+    this.datasourceSubscription.unsubscribe();
+  }
+
+  applyFilter(event: Event): void {
+    const filterValue = (event.target as HTMLInputElement).value;
+    this.datasource.filter = filterValue.trim();
+  }
+
+  applyPaginationSize(sizeValue: string): void {
+    this.datasource.options = { size: Number.parseInt(sizeValue, 10) };
```

Comment:
Why is parse int used instead of `+`, is it becouse part of it could be text?
```suggestion
    this.datasource.options = { size: +sizeValue };
```

Reply:
If its is not parsable it now will use the default vaue `10`

Reply:
I don;t think that field is for a default